docs/standard/serialization/system-text-json/populate-properties.md

```diff
@@ -0,0 +1,62 @@
+    public List<int> Numbers2 { get; set; } = new List<int>() { 1, 2, 3 };
+}
+```
+
+When you create an instance of this class, the `Numbers1` (and `Numbers2`) property's value is a list with three elements (1, 2, and 3). If you deserialize JSON to this type, the default behavior is that property values are replaced:
+
+- For `Numbers1`, since it's read-only (no setter), it will still have the values 1, 2, and 3 in its list.
+- For `Numbers2`, which is read-write, the contents of the list will be replaced with the values from the JSON.
+
+For example, if you execute the following deserialization code, `Numbers1` contains the values 1, 2, and 3 and `Numbers2` contains the values 4, 5, and 6.
+
+```csharp
+A? a = JsonSerializer.Deserialize<A>("""{"Numbers1": [4,5,6], "Numbers2": [4,5,6]}""");
+```
+
+Starting in .NET 8, you can change the deserialization behavior to modify (*populate*) initialized properties and fields instead of replacing them. And since the value that the property references isn't *replaced* (that is, doesn't require a setter), if the type of the property is mutable, deserialization can also populate *read-only* properties.
+
+The <xref:System.Text.Json.Serialization.JsonObjectCreationHandling> enum fields provide the object creation handling choices:
+
+- <xref:System.Text.Json.Serialization.JsonObjectCreationHandling.Replace?displayProperty=nameWithType> (matches the default behavior)
+- <xref:System.Text.Json.Serialization.JsonObjectCreationHandling.Populate?displayProperty=nameWithType>
+
+There are multiple ways to specify a preference for *replace* or *populate*:
+
+- Use the <xref:System.Text.Json.Serialization.JsonObjectCreationHandlingAttribute> attribute to annotate at the type or property level. If you set the attribute at the type level and set its <xref:System.Text.Json.Serialization.JsonObjectCreationHandlingAttribute.Handling> property to <xref:System.Text.Json.Serialization.JsonObjectCreationHandling.Populate>, the populate behavior will only apply to those properties where it's possible (for example, value types must have a setter).
+
+  ```csharp
+  [JsonObjectCreationHandling(JsonObjectCreationHandling.Populate)]  // Type-level
+  class A
+  {
+      [JsonObjectCreationHandling(JsonObjectCreationHandling.Populate)]  // Property-level
+      public List<int> Numbers1 { get; } = new List<int>() { 1, 2, 3 };
+      public List<int> Numbers2 { get; set; } = new List<int>() { 1, 2, 3 };
+  }
+  ```
+
+- Set <xref:System.Text.Json.JsonSerializerOptions.PreferredObjectCreationHandling?displayProperty=nameWithType> (or, for source generation, <xref:System.Text.Json.Serialization.JsonSourceGenerationOptionsAttribute.PreferredObjectCreationHandling?displayProperty=nameWithType>) to specify a global preference.
+
+  ```csharp
+  var options = new JsonSerializerOptions
+  {
+      PreferredObjectCreationHandling = JsonObjectCreationHandling.Populate
+  };
```

Comment:
Would it be possible to also include a few examples from the `Property/Field type description` table as specified in https://github.com/dotnet/runtime/issues/78556?

It probably doesn't need to be as expensive, but we should maybe document how the feature interacts with properties whose types are structs (i.e. populate won't work unless unless the struct property includes a setter).

Reply:
I don't think the struct property needs to have a setter for populate to work - I tried the following struct and both lists ended up with all six values (1-6).

```csharp
S? s = JsonSerializer.Deserialize<S>("""{"Numbers1": [4,5,6], "Numbers2": [4,5,6]}""");

struct S
{
    [JsonObjectCreationHandling(JsonObjectCreationHandling.Populate)]
    public S() { }

    public List<int> Numbers1 { get; } = new List<int>() { 1, 2, 3 };
    public List<int> Numbers2 { get; set; } = new List<int>() { 1, 2, 3 };
}
```